src/pages/index.js

```diff
@@ -68,38 +52,199 @@ const profileEditModal = new PopupWithForm(
 
+const deleteImageModal = new PopupConfirmDelete("#confirm-delete-modal");
+deleteImageModal.setEventListeners();
+
+const avatarModal = new PopupWithForm(
+  {
+    popupSelector: "#change-avatar-modal",
+  },
+  handleChangeAvatar
+);
+avatarModal.setEventListeners();
+
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+api
+  .getCurrentUserInfo()
+  .then((userData) => {
+    userInfo.setAvatarInfo(userData);
+    userInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      ".cards__list"
+    );
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
 function handleProfileEditSubmit(inputValues) {
-  console.log(inputValues);
-  userInfo.setUserInfo(inputValues);
-  editFormValidator.resetValidation(inputValues);
-  profileEditModal.close();
+  profileEditModal.renderLoading(true);
+  api
+    .updateProfileInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      profileEditModal.close();
+      editFormElement.resetValidation();
+      editFormValidator.disableSubmit();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditModal.renderLoading(false);
+    });
 }
 
 function handleAddCardFormSubmit(inputValues) {
-  renderCard(inputValues);
-  addFormValidator.resetValidation(inputValues);
+  addCardModal.renderLoading(true);
+  api
+    .createACard(inputValues)
+    .then((res) => {
+      renderCard(res);
+      addCardModal.close();
+      addFormElement.resetValidation();
+      addFormValidator.disableSubmit();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardModal.renderLoading(false);
+    });
+}
 
-  addCardModal.close();
+function handleChangeAvatar(inputValues) {
+  avatarModal.renderLoading(true);
+  api
+    .updateAvatar(inputValues)
+    .then((res) => {
+      userInfo.setAvatarInfo(res);
+      avatarModal.close();
+      avatarFormValidator.disableSubmit();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarModal.renderLoading(false);
+    });
+  console.log(inputValues);
 }
 
-const editFormValidator = new FormValidator(
-  validationSettings,
-  editFormElement
-);
-const addFormValidator = new FormValidator(validationSettings, addFormElement);
+function handleImageClick(cardData) {
+  previewImageModal.open(cardData);
+}
 
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
+function handleLikeClicks(card) {
+  if (card.isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then((newCardData) => {
+        card.isLiked = newCardData.isLiked;
```

Comment:
Needs correction: please don't change the instance variables directly,. create a setter